zeam-vm/pelemay_backend
# refactor(OnnxToAxonBench): factor out http utils

### Description

Part of the effort toward https://github.com/zeam-vm/pelemay_backend/issues/72

This pull request factors HTTP-related utilities out of `OnnxToAxonBenc` module. At the high level, no existing behavior is changed. 

To test

```
cd path/to/pelemay_backend/benchmarks/onnx_to_axon_bench

mix format --check-formatted
mix credo
mix dialyzer
mix run -e "OnnxToAxonBench.run"
```

✅ ALL GOOD! (other than CI...)

### Notes
- I replace [`Req.url()`](https://hexdocs.pm/req/0.3.9/Req.html#types) type with more generic `URI.t() | String.t()` so that our interface won't depend on the [req](https://hex.pm/packages/req) package
- I name `OnnxToAxonBench.Utils.HTTP` after [`Bumblebee.Utils.HTTP`](https://github.com/elixir-nx/bumblebee/tree/main/lib/bumblebee/utils) but it can be a different name if we come up with a better one for our context
- We could `alias OnnxToAxonBench.Utils.HTTP` but here I choose to `import` so that I do not need to touch any logic within `OnnxToAxonBench`
